tests/test_unit/test_cdf_tk/test_resource_classes/test_dataset.py

```diff
@@ -0,0 +1,30 @@
+import pytest
+
+from cognite_toolkit._cdf_tk.constants import MODULES
+from cognite_toolkit._cdf_tk.resource_classes import DataSetYAML
+from cognite_toolkit._cdf_tk.utils.file import read_yaml_file
+from tests.data import COMPLETE_ORG
+
+
+def find_resources(resource: str):
+    base = COMPLETE_ORG / MODULES
+    for path in base.rglob(f"*{resource}.yaml"):
+        data = read_yaml_file(path)
+        if isinstance(data, dict):
+            yield pytest.param(data, id=path.relative_to(base).as_posix())
+        elif isinstance(data, list):
+            for no, item in enumerate(data):
+                if isinstance(item, dict):
+                    yield pytest.param(item, id=f"{path.relative_to(base).as_posix()} - Item: {no}")
+                else:
+                    raise ValueError(f"Invalid data format in {path}: {item}")
+        else:
+            raise ValueError(f"Invalid data format in {path}: {data}")
```

Comment:
Lets move this out to a utility function and update the test_timeseries to use the same function